scripts/parse_modules.py

```diff
@@ -0,0 +1,135 @@
+import ast, os, copy
+import yaml
+
+
+__all__ = ["api_crawler", "differ"]
+
+
+def differ(old_version, new_version):
+    with open(old_version, "r") as f:
+        old = f.read()
+        old = yaml.load(old)
+    with open(new_version, "r") as f:
+        new = f.read()
+        new = yaml.load(new)
+    combined = copy.deepcopy(old)
+    combined.update(new)
+
+    diff = {}
+    union = {}
+    files_union = set(old) & set(new)
+    files_diff = set(old) - set(new)
+
+    # Diff files
+    for x in combined.keys():
+        if x in files_diff:
+            diff[x] = combined[x]
+            diff[x] = {}
+        else:
+            union[x] = combined[x]
+
+    # Diff functions and classes
+    for x in union.keys():
+        old_items = old.get(x, None)
+        new_items = new.get(x, None)
+        if old_items and new_items:
+            function_diff = set(old_items["functions"]) - set(new_items["functions"])
+            class_diff = set(list(old_items["classes"].keys())) - set(list(new_items["classes"].keys()))
+            if function_diff or list(class_diff):
+                diff[x]= copy.deepcopy(union[x])
+                assert(id(diff[x]) != id(union[x]))
+                if list(class_diff):
+                    diff_dict = {y: {"methods": []} for y in class_diff}
+                    diff[x]["classes"] = diff_dict
+                else:
+                    diff[x]["classes"] = {}
+
+                if function_diff:
+                    diff[x]["functions"] = list(function_diff)
+                else:
+                    diff[x]["functions"] = []
+
+    # Diff methods
+    for x in union.keys():
+        old_classes = old[x].get("classes", None) if old.get(x, None) else {}
+        new_classes = new[x].get("classes", None) if new.get(x, None) else {}
+        if old_classes and new_classes:
+            pass
+
+    return diff
+
+
+class APICrawler(object):
+    exclude = ("tests", "static", "sampledata", "mplexplorer")
+
+    def __init__(self, directory):
+        self.directory = directory
+
+    def is_public(self, name):
+        if name.startswith("_", 0, 1) and name != "__init__":
+            return False
+        else:
+            return True
+
+    def is_function(self, ast_node):
+        if isinstance(ast_node, ast.FunctionDef) and ast_node.col_offset == 0:
+            return True
+        else:
+            return False
+
+    def is_class(self, ast_node):
+        if isinstance(ast_node, ast.ClassDef):
+            return True
+        else:
+            return False
```

Comment:
Simple `return bool_expression` or `return not (bool_expression)` (depending on the order of cases) is sufficient in all three methods.